src/ray/gcs/gcs_server/test/gcs_server_rpc_test.cc

```diff
@@ -51,74 +50,100 @@ class GcsServerTest : public RedisServiceManagerForTest {
     thread_gcs_server_->join();
   }
 
-  void TestAddJob(const rpc::AddJobRequest &request) {
-    auto call_function = [this, request](std::promise<bool> &promise) {
-      client_->AddJob(request,
-                      [&promise](const Status &status, const rpc::AddJobReply &reply) {
-                        RAY_CHECK_OK(status);
-                        promise.set_value(true);
-                      });
-    };
-    AsyncCall(call_function, timeout_ms_);
+  bool AddJob(const rpc::AddJobRequest &request) {
+    std::promise<bool> promise;
+    client_->AddJob(request, [&](const Status &status, const rpc::AddJobReply &reply) {
```

Comment:
```suggestion
    client_->AddJob(request, [&promise](const Status &status, const rpc::AddJobReply &reply) {
```
Explicitly list the captures.

Reply:
fixed